vunit/ui/results.py

```diff
@@ -24,5 +29,76 @@ def merge_coverage(self, file_name, args=None):
         :param file_name: The resulting coverage file name.
         :param args: The tool arguments for the merge command. Should be a list of strings.
         """
-
         self._simulator_if.merge_coverage(file_name=file_name, args=args)
+
+    def get_report(self):
+        """
+        Get a report (dictionary) of tests: status, time and output path
+
+        :returns: A :class:`Report` object
+        """
+        report = Report(self._output_path)
+        for (
+            test
+        ) in self._report._test_results_in_order():  # pylint: disable=protected-access
+            obj = test.to_dict()
+            report.tests.update(
+                {test.name: ReportResult(obj["status"], obj["time"], obj["path"])}
+            )
+        return report
+
+
+class Report(object):
+    """
+    Gives access to test results and paths after running tests
+
+    :data output_path: Absolute path to the output path (see :ref:`cli`)
+    :data tests: Dictionary of :class:`ReportResult` objects
+    """
+
+    def __init__(self, output_path):
+        self.output_path = output_path
+        self.tests = {}
+
+    def get_relpath(self, path):
```

Comment:
I still think this should be property of the individual tests result since the path is associated with an individual test. The fact that the test object needs a private copy of the output path is no reason not to do it.